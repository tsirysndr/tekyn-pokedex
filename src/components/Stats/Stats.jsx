import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "@emotion/native";
import { View } from "react-native";
import DefaultText from "@Components/Text";
import { useFonts, Roboto_700Bold } from "@expo-google-fonts/roboto";
import _ from "lodash";

const Container = styled.View`
  height: ${(props) => props.height}px;
  width: ${(props) => props.width}px;
  background-color: #c4c4c4;
  padding: 4px;
  padding-bottom: 0;
`;

const Cell = styled.View`
  flex: 1;
  border-bottom-width: 0.5px;
  border-bottom-color: #c4c4c4;
  border-right-width: 0.2px;
  border-right-color: #c4c4c4;
  ${({ active }) =>
    active &&
    css`
      background-color: #38a8d5;
    `}
`;

const Indicators = styled.View`
  flex-direction: row;
`;

const Text = styled(DefaultText)`
  font-size: 6px;
  ${({ fontsLoaded }) =>
    fontsLoaded &&
    css`
      font-family: "Roboto_700Bold";
    `}
`;

const Centered = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  height: 10px;
`;

const WhiteBrackground = styled.View`
  flex: 1;
  background-color: #fff;
  flex-direction: row;
`;

const Level = ({ value }) => {
  return (
    <View style={{ flex: 1 }}>
      <Cell active={value >= 15}></Cell>
      <Cell active={value >= 14}></Cell>
      <Cell active={value >= 13}></Cell>
      <Cell active={value >= 12}></Cell>
      <Cell active={value >= 11}></Cell>
      <Cell active={value >= 10}></Cell>
      <Cell active={value >= 9}></Cell>
      <Cell active={value >= 8}></Cell>
      <Cell active={value >= 7}></Cell>
      <Cell active={value >= 6}></Cell>
      <Cell active={value >= 5}></Cell>
      <Cell active={value >= 4}></Cell>
      <Cell active={value >= 3}></Cell>
      <Cell active={value >= 2}></Cell>
      <Cell active={value >= 1}></Cell>
    </View>
  );
};

Level.defaultProps = {
  value: 0,
};

Level.propTypes = {
  value: PropTypes.number,
};

export default function Stats(props) {
  const [fontsLoaded] = useFonts({ Roboto_700Bold });
  const computeValue = (x) => Math.floor(_.get(x, "base_stat", 0) / 10);
  return (
    <Container {...props}>
      <WhiteBrackground>
        <Level
          value={computeValue(
            props.pokemonstats.find((x) => x["pokemon_v2_stat"].name === "hp")
          )}
        />
        <Level
          value={computeValue(
            props.pokemonstats.find(
              (x) => x["pokemon_v2_stat"].name === "attack"
            )
          )}
        />
        <Level
          value={computeValue(
            props.pokemonstats.find(
              (x) => x["pokemon_v2_stat"].name === "defense"
            )
          )}
        />
        <Level
          value={computeValue(
            props.pokemonstats.find(
              (x) => x["pokemon_v2_stat"].name === "special-attack"
            )
          )}
        />
        <Level
          value={computeValue(
            props.pokemonstats.find(
              (x) => x["pokemon_v2_stat"].name === "special-defense"
            )
          )}
        />
        <Level
          value={computeValue(
            props.pokemonstats.find(
              (x) => x["pokemon_v2_stat"].name === "speed"
            )
          )}
        />
      </WhiteBrackground>
      <Indicators>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            HP
          </Text>
        </Centered>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            Atack
          </Text>
        </Centered>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            Defense
          </Text>
        </Centered>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            Spe A
          </Text>
        </Centered>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            Spe D
          </Text>
        </Centered>
        <Centered>
          <Text fontsLoaded={fontsLoaded} color="#000">
            Speed
          </Text>
        </Centered>
      </Indicators>
    </Container>
  );
}

Stats.defaultProps = {
  height: 215,
  width: 346,
  pokemonstats: [],
};

Stats.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
  pokemonstats: PropTypes.array.isRequired,
};
