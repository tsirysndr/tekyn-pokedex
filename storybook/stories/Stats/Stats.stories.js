import React from "react";
import Stats from "@Components/Stats";
import { storiesOf } from "@storybook/react-native";
import CenterView from "../CenterView";
import { pokemonstats } from "../../../__mock__";

storiesOf("Stats", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("empty", () => <Stats />)
  .add("with data", () => <Stats pokemonstats={pokemonstats} />);
