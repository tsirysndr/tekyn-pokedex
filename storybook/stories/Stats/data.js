export const pokemonstats = [
  {
    base_stat: 50,
    effort: 0,
    id: 943,
    pokemon_id: 158,
    stat_id: 1,
    pokemon_v2_stat: { name: "hp" },
  },
  {
    base_stat: 65,
    effort: 1,
    id: 944,
    pokemon_id: 158,
    stat_id: 2,
    pokemon_v2_stat: { name: "attack" },
  },
  {
    base_stat: 64,
    effort: 0,
    id: 945,
    pokemon_id: 158,
    stat_id: 3,
    pokemon_v2_stat: { name: "defense" },
  },
  {
    base_stat: 44,
    effort: 0,
    id: 946,
    pokemon_id: 158,
    stat_id: 4,
    pokemon_v2_stat: { name: "special-attack" },
  },
  {
    base_stat: 48,
    effort: 0,
    id: 947,
    pokemon_id: 158,
    stat_id: 5,
    pokemon_v2_stat: { name: "special-defense" },
  },
  {
    base_stat: 43,
    effort: 0,
    id: 948,
    pokemon_id: 158,
    stat_id: 6,
    pokemon_v2_stat: { name: "speed" },
  },
];
