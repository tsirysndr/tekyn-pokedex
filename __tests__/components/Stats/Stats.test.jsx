import Stats from "@Components/Stats";
import { render } from "@testing-library/react-native";

test("renders Stats", () => {
  const tree = render(
    <Stats />
  );
  expect(tree).toMatchSnapshot();
});
