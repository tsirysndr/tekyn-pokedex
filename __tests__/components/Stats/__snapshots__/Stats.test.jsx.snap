// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stats 1`] = `
<View
  height={215}
  pokemonstats={Array []}
  style={
    Array [
      Object {
        "backgroundColor": "#c4c4c4",
        "paddingBottom": 0,
        "paddingLeft": 4,
        "paddingRight": 4,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  width={346}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": 10,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
